### PR TITLE
fix(*): creating and using interfaces for system tasks

### LINKED ISSUE #227: Increase test coverage in ./pkg/gitreceive/storage
Several important `func`s are untested because they interact with the filesystem or environment. Here's a non-exhaustive list:

- `GetClient`
- `getEndpoint`
- `getAuth`
- `CredsOK` (relies on `getAuth`)

In order to test them, put the I/O behind interfaces that can be mocked.

If #226 is done before this, the storage package will be moved to `./pkg/storage`, but this issue remains otherwise the same.

### PR DESCRIPTION
**Pull request by arschles**
This PR creates interfaces meant for interacting with environment variables and for reading from the filesystem. Various pieces of the storage package use these interfaces, and this PR also adds mock implementations of those interfaces and unit tests for the aforementioned pieces.

Fixes #227 

TODO: 

- [x] Create interfaces, real and mock implementations for system tasks
- [x] Add tests for functionality that now uses the aforementioned interfaces

### CODE REVIEW

File: pkg/gitreceive/build.go [hunk 1]
```diff
@@ -41,7 +42,7 @@ func run(cmd *exec.Cmd) error {
 	return cmd.Run()
 }
 
-func build(conf *Config, s3Client *s3.S3, kubeClient *client.Client, builderKey, rawGitSha string) error {
+func build(conf *Config, s3Client *s3.S3, kubeClient *client.Client, env sys.Env, fs sys.FS, builderKey, rawGitSha string) error {
```
Comment (jackfrancis) [inline pkg/gitreceive/build.go]: This is ticky-tacky, but... all other functions that accept both a sys.FS and sys.ENV interface do so in according to an ordered arity that looks like `(<params>...sys.FS, sys.ENV,...<params>)`. Maybe switch the order of `env sys.Env` and `fs sys.FS` params in `build()` above.

Comment (arschles) [inline pkg/gitreceive/build.go]: @jackfrancis fair point, change incoming

**Comment by jackfrancis** [comment]
Looks great, interfaces ftw.

**Comment by smothiki** [comment]
More interfaces awsome

**Pull request merged**